aws/resource_aws_organizations_account_test.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func testAccAwsOrganizationsAccount_basic(t *testing.T) {
+	var account organizations.Account
+
+	test_email, ok := os.LookupEnv("TEST_AWS_ORGANIZATION_ACCOUNT_EMAIL")
+
+	if !ok {
+		t.Skip("'TEST_AWS_ORGANIZATION_ACCOUNT_EMAIL' not set, skipping test.")
+	}
+
+	name := "my_new_account"
+	email := test_email
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsOrganizationsAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsOrganizationsAccountConfig(name, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsOrganizationsAccountExists("aws_organizations_account.test", &account),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsOrganizationsAccountDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).organizationsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_organizations_account" {
+			continue
+		}
+
+		params := &organizations.DescribeAccountInput{
+			AccountId: &rs.Primary.ID,
+		}
+
+		resp, err := conn.DescribeAccount(params)
+
+		if err != nil {
+			return err
```

Comment:
We might need to check for `AccountNotFoundException` as the SDK documentation notes:

> We can't find an AWS account with the AccountId that you specified, or the
account whose credentials you used to make this request is not a member of
an organization.

I believe in our testing at this point the testing account will be removed from the organization again, so this would be the case:

```go
if isAWSErr(err, organizations.ErrCodeAccountNotFoundException, "") {
  return nil
}
```